common/src/main/java/com/microsoft/identity/common/internal/ui/embeddedwebview/EmbeddedWebViewAuthorizationStrategy.java

```diff
@@ -0,0 +1,125 @@
+ */
+public class EmbeddedWebViewAuthorizationStrategy extends AuthorizationStrategy {
+    private static final String TAG = StringExtensions.class.getSimpleName();
+    private WebView mWebView;
+    private String mStartUrl;
+
+    /**
+     * Perform the authorization request in the embedded web view and return the authorization result.
+     *
+     * @param request authorization request
+     * @return AuthorizationResult
+     */
+    public AuthorizationResult requestAuthorization(@NonNull final AuthorizationRequest request) {
+        loadStartUrl();
+        throw new UnsupportedOperationException("Not implemented yet.");
+    }
+
+    /**
+     * Constructor of EmbeddedWebViewAuthorizationStrategy.
+     *
+     * @param activity Authentication activity
+     * @param request  Authorization request
+     * @throws UnsupportedEncodingException
+     * @throws ClientException
+     */
+    public EmbeddedWebViewAuthorizationStrategy(@NonNull final Activity activity,
+                                                @NonNull final AuthorizationRequest request)
+            throws UnsupportedEncodingException, ClientException {
+        //TODO validate auth request in OAuth2Strategy.
+        createWebView(activity, request);
+        mStartUrl = request.getAuthorizationStartUrl();
+    }
+
+    /**
+     * Set up the web view configurations.
+     * @param activity  AuthenticationActivity
+     * @param request AuthorizationRequest
+     */
+    private void createWebView(final Activity activity, final AuthorizationRequest request) {
+        // Create the Web View to show the page
+        mWebView = (WebView) activity.findViewById(activity.getResources().getIdentifier("webView1", "id",
+                activity.getPackageName()));
+        mWebView.getSettings().setUserAgentString(
+                mWebView.getSettings().getUserAgentString() + AuthenticationConstants.Broker.CLIENT_TLS_NOT_SUPPORTED);
+        mWebView.getSettings().setJavaScriptEnabled(true);
+        mWebView.requestFocus(View.FOCUS_DOWN);
+
+        // Set focus to the view for touch event
+        mWebView.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(final View view, final MotionEvent event) {
+                int action = event.getAction();
+                if ((action == MotionEvent.ACTION_DOWN || action == MotionEvent.ACTION_UP) && !view.hasFocus()) {
+                    view.requestFocus();
+                }
+                return false;
+            }
+        });
+
+        mWebView.getSettings().setLoadWithOverviewMode(true);
+        mWebView.getSettings().setDomStorageEnabled(true);
+        mWebView.getSettings().setUseWideViewPort(true);
+        mWebView.getSettings().setBuiltInZoomControls(true);
+        mWebView.setVisibility(View.INVISIBLE);
+        mWebView.setWebViewClient(new AzureActiveDirectoryWebViewClient(activity, request));

```

Comment:
Can we make `AzureActiveDirectoryWebViewClient` the parameter to this method instead of `AuthorizationRequest`?

Furthermore, can we make the `AzureActiveDirectoryWebViewClient` a parameter to the constructor of `EmbeddedWebViewAuthorizationStrategy`?

If we don't do this, we cannot mock this object in test

Reply:
Addressed the first part. But I don't think we should take the webviewclient as a parameter for EmbeddedWebViewAuthorizationStrategy constructor. The usage/logic of webviewclient is within the webview. We should not expose it out of the authorization strategy. Authorization strategy should only take the 1) `authorization request` and 2) `the activity which to perform this request` as the input.

Reply:
How are you going to mock this then?

Reply:
It is almost always preferable to pass-in your dependencies, rather than construct them yourself -- this is the purpose of dependency injection.

If this is for AAD only, wouldn't it make more sense to call this `AzureActiveDirectoryEmbeddedWebViewAuthorizationStrategy`?

Reply:
The interaction between `EmbeddedWebViewAuthorizationStrategy` and `AzureActiveDirectoryWebViewClient` should exist as a interface anyway, so even more reason to make this change